migration/migration.go

```diff
@@ -0,0 +1,68 @@
+package migration
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/NethermindEth/juno/db"
+)
+
+type revision func(db.DB) error
+
+var revisions = []revision{
+	revision0000,
+}
+
+func MigrateIfNeeded(targetDB db.DB) error {
+	version, err := SchemaVersion(targetDB)
+	if err != nil {
+		return err
+	}
+
+	for i := version; i < uint64(len(revisions)); i++ {
+		if err = revisions[i](targetDB); err != nil {
+			return err
+		}
+
+		// revision returned with no errors, bump the version
+		if err = targetDB.Update(func(txn db.Transaction) error {
+			var versionBytes [8]byte
+			binary.BigEndian.PutUint64(versionBytes[:], i+1)
+			return txn.Set(db.SchemaVersion.Key(), versionBytes[:])
+		}); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func SchemaVersion(targetDB db.DB) (uint64, error) {
```

Comment:
This doesn't appear to be used outside the `migration` package. Could it be unexported?

Reply:
In that case, could the tests be moved to a separate package?

Reply:
Yeap, it should've been in `migration_test` in the first place. My bad.